addons/sale_timesheet/controllers/portal.py

```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.osv import expression
+
+from odoo.addons.hr_timesheet.controllers.portal import CustomerPortal
+
+
+class CustomerPortal(CustomerPortal):
+
+    def _get_timesheet_domain(self):
+        domain = super(CustomerPortal, self)._get_timesheet_domain()
+        domain = expression.AND([domain, [('timesheet_invoice_type', 'in', ['billable_time', 'non_billable']), ('so_line', '=', False)]])
+        return expression.OR([domain, [('timesheet_invoice_id', '!=', False)]])
```

Comment:
`('timesheet_invoice_id', '!=', False)]` should be added to portal domain, to search timesheet linked on the current invoice (in account customer portal, probably in `_invoice_get_page_view_values`).
So, like for task, don't use `invoice.timesheet_ids`, but add 'timesheet' in values to render template.